source/distributed/shared_tria.cc

```diff
@@ -134,6 +135,20 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &old_tria)
+    {
+      Assert ((dynamic_cast<const dealii::parallel::distributed::Triangulation<dim,spacedim> *>(&old_tria) == NULL),
+              ExcMessage("Cannot use this function on parallel::distributed::Triangulation."));
```

Comment:
I think the assertion should be that the argument *is* a `p::s::Triangulation`, not that it may be something else.